hazelcast/src/main/java/com/hazelcast/map/impl/MapKeyLoader.java

```diff
@@ -414,6 +417,10 @@ private void sendKeysInBatches(MapStoreContext mapStoreContext, boolean replaceE
 
         try {
             Iterable<Object> allKeys = mapStoreContext.loadAllKeys();
+            // if no keys to load
+            if (allKeys == Collections.emptyList()) {
```

Comment:
Our implementations of `mapStoreContext.loadAllKeys()` indeed return `Collections.emptyList()` at the moment, but I think checking the iterator instead of using `==` would be safer / more future-proof.